### PR TITLE
feat: setup base Rails app

### PR DESCRIPTION
**Pull request by evfinkn**
Use `--minimal` flag since things like `ActionMailer` likely won't be useful for our app and can always be added later.

### CODE REVIEW

**Comment by evfinkn** [comment]
In hindsight, `ActionMailer` actually could be useful for emailing other users, but again, we shouldn't add it until it's needed.

File: Gemfile
Comment (powersaudrey25) [inline Gemfile:1]: I believe we will need to specify to use pg for heroku. 
group :production do
  gem 'pg' # for Heroku deployment
  gem 'rails_12factor'
end

Comment (evfinkn) [reply Gemfile:1]: I just read [Heroku's Rails 7 guide](https://devcenter.heroku.com/articles/getting-started-with-rails7), and they strongly recommend using Postgres everywhere:

> Heroku highly recommends using PostgreSQL locally during development. Maintaining parity between development and deployment environments prevents introducing subtle bugs due to the differences in environments.

Do we think we should use Postgres in production or in every environment?

Also, we talked about it in the meeting, but for future reference: `rails_12factor` isn't needed for Rails 5.x and later.

Comment (powersaudrey25) [reply Gemfile:1]: We can try it. I don't think the implementation will be any different.

Comment (kesslerjohn) [reply Gemfile:1]: I agree - I don't see any downsides to using Postgres in every environment as long as everyone's local environment supports it

**Comment by evfinkn** [comment]
For `pre-commit` to work, I had to do `rbenv global 3.1.3` since `pre-commit` defaults to the system Ruby version. See [pre-commit issue #1930](https://www.github.com/pre-commit/pre-commit/issues/1930#issuecomment-852169116):

> ruby defaults to using your system installation (for speed, and for windows)

Specifying `language_version` installs an entire new Ruby instance which isn't ideal (and that quote makes it seem like it wouldn't work for Windows?). Is this okay with everyone?

**Comment by powersaudrey25** [comment]
> For `pre-commit` to work, I had to do `rbenv global 3.1.3` since `pre-commit` defaults to the system Ruby version. See [pre-commit issue #1930](https://www.github.com/pre-commit/pre-commit/issues/1930#issuecomment-852169116):
> 
> > ruby defaults to using your system installation (for speed, and for windows)
> 
> Specifying `language_version` installs an entire new Ruby instance which isn't ideal (and that quote makes it seem like it wouldn't work for Windows?). Is this okay with everyone?

I think this is okay. If I am understanding correctly, we just have to make sure our global version of ruby is 3.1.3 for pre-commit to work?

**Review by powersaudrey25 — APPROVED** [review]

**Review by powersaudrey25 — CHANGES_REQUESTED** [review]
Oh wait, we also need the SimpleCov gem I don't see it in the gemfile

**Comment by evfinkn** [comment]
> > For `pre-commit` to work, I had to do `rbenv global 3.1.3` since `pre-commit` defaults to the system Ruby version. See [pre-commit issue #1930](https://www.github.com/pre-commit/pre-commit/issues/1930#issuecomment-852169116):
> > > ruby defaults to using your system installation (for speed, and for windows)
> > 
> > 
> > Specifying `language_version` installs an entire new Ruby instance which isn't ideal (and that quote makes it seem like it wouldn't work for Windows?). Is this okay with everyone?
> 
> I think this is okay. If I am understanding correctly, we just have to make sure our global version of ruby is 3.1.3 for pre-commit to work?

Correct (in addition to installing it). I added `pre-commit` installation instructions and PostgreSQL installation instructions for macOS with Homebrew to the README.

**Review by powersaudrey25 — APPROVED** [review]

**Pull request merged**